### PR TITLE
add publish job to ci

### PR DESCRIPTION
**Pull request by mythmon**
This is nearly a direct copy of [GitHub's documentation](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages) on publishing images to the GitHub image registry.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
```
Comment (mootari) [inline .github/workflows/publish.yml:23]:
```suggestion
        uses: docker/login-action@e92390c5fb421da1463c202d546fed0ec5c39f20 # v3.1.0
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Extract metadata for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
```
Comment (mootari) [inline .github/workflows/publish.yml:30]:
```suggestion
        uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81 # v5.5.1
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+      - name: Build and push Docker image
+        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
```
Comment (mootari) [inline .github/workflows/publish.yml:34]:
```suggestion
        uses: docker/build-push-action@2cdde995de11925a030ce8070c3d77a52ffcf1c0 # v5.3.0
```

**Review by mootari — APPROVED** [review]
I would recommend to update to the most recent releases, as the ones from the example appear to be from March '23.

**Comment by mythmon** [comment]
Thanks for checking that.

**Pull request merged**